### PR TITLE
Updated UI of No Match and Coming Soon pages

### PR DESCRIPTION
**Pull request by MVarshini**
PBENCH -989

Empty State for Upcoming Pages

### CODE REVIEW

**Review by dbutenhof — DISMISSED** [review]
It's not a big deal, but I'm curious why we get treated to a set of images (including a Pbench logo) for the 404 error but "coming soon" is just text? 😆

**Comment by dbutenhof** [comment]
> /home/jenkins/pbench/dashboard/src/assets/images/404-Error-Not-found.png
  1:1  error  Parsing error: Unexpected character '�'

> ✖ 1 problem (1 error, 0 warnings)

File: dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import "./index.less";
+
+import { Button } from "@patternfly/react-core";
+import React from "react";
+import logo from "assets/images/404-Error-Not-found.png";
```
Comment (portante) [inline dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx:5]: Perhaps call this "page_not_found".

**Review by portante — CHANGES_REQUESTED** [review]

File: dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx (hunk 1 above)
Comment (webbnh) [reply dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx:5]: Yes, we need a hyphen that is shaped like a comma.  😆  It's not "Hey, we didn't find the 404-Error", it's "Hey it's the not-found error, 404".  🤣

Seriously, though, Peter's is a good solution:  this is a case where "less is more" -- having "404" _and_ "error" _and_ "not found" all in the name makes things less clear with more room for silliness.

File: dashboard/src/modules/components/EmptyPageComponent/ComingSoon.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import React from "react";
+import { useNavigate } from "react-router-dom";
+
+const ComingSoonPage = () => {
+  const navigate = useNavigate();
+  return (
+    <div className="comingsoon-page-container">
+      <h1>
+        Coming Soon<span className="dot">.</span>
+      </h1>
```
Comment (webbnh) [inline dashboard/src/modules/components/EmptyPageComponent/ComingSoon.jsx:13]: Wow:  what does the dot do if we _don't_ wrap it in a `<span>`??  😀

**Review by webbnh — APPROVED** [review]
I like the elk on the 404 page -- very retro.  😀

**Comment by webbnh** [comment]
> > /home/jenkins/pbench/dashboard/src/assets/images/404-Error-Not-found.png
> > 1:1  error  Parsing error: Unexpected character '�'
> 
> > ✖ 1 problem (1 error, 0 warnings)

We've told ESLint to ignore `.ico`, `.jpg`, and `.svg` files, but we didn't tell it to ignore `.png` files.  🥴 

So, @MVarshini, you should update `dashboard/.eslintignore` here, as well.

**Comment by webbnh** [comment]
And...now that #3082 has been merged, you need to rebase and resolve the conflict in `App.js`.

File: dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx (hunk 1 above)
Comment (MVarshini) [reply dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx]: Renamed the image :grinning:

**Review by webbnh — DISMISSED** [review]
👍

**Review by dbutenhof — DISMISSED** [review]

**Review by siddardh-ra — DISMISSED** [review]

**Review by portante — APPROVED** [review]

**Comment by webbnh** [comment]
Mr. Jenkins (well, OK, `eslint`, actually) requests some changes to the  `alt` props of two of your `img`'s:
```
/home/jenkins/pbench/dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx
  14:9   warning  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images                                                                                        jsx-a11y/alt-text
  24:13  warning  Redundant alt attribute. Screen-readers already announce `img` tags as an image. You don’t need to use the words `image`, `photo,` or `picture` (or any specified custom words) in the alt prop  jsx-a11y/img-redundant-alt
```

File: dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx [hunk 3]
```diff
@@ -21,7 +22,11 @@ const NoMatchingPage = () => {
       <div className="nomatch-container">
         <div className="main-content">
           <div className="bg-img-wrapper">
-            <img className="not-found-img" alt="not found image" src={logo} />
+            <img
+              className="not-found-img"
+              alt="elk with page not found message"
```
Comment (webbnh) [inline dashboard/src/modules/components/EmptyPageComponent/NoMatchingPage.jsx:27]: 😆  I love that the alt message includes the "elk"!

**Review by webbnh — APPROVED** [review]
The changes look fine.  Unfortunately, one of the Pbench Server functional tests failed, which (presumably) has nothing to do with this PR.

There's a ginormous traceback (or multiple tracebacks...it's really hard to tell) which seems to center around `psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "templates_name_key"`, which is I believe a timing-based failure that Dave is working on.  So, I'll resubmit the job and see if it passes this time.

Hey @dbutenhof, is this "S3 upload" failure expected?  (It doesn't seem to be causing any problems, it just jumped out at me as I was looking at the log, so I thought I'd ask.)
```
Dec 09 12:59:04 pbenchinacan pbench-backup-t[432]: 2022-12-09T12:59:04.531500 INFO 432 139667429432192 pbench-backup-tarballs pbench-backup-tarballs main 458 -- Total processed: 5, Local backup successes: 5, Local backup failures: 0, S3 upload successes: 0, S3 upload failures: 5, Unable to process: 0
```
I see this message periodically in the log, so I assume it explains the previous:
```
Dec 09 13:08:01 pbenchinacan pbench-backup-t[729]: 2022-12-09T13:08:01.894711 WARNING 729 140579956992896 pbench-backup-tarballs pbench-backup-tarballs sanity_check 71 -- Bucket not defined in config file - S3 backup is disabled.
```
But, it's interesting that we note a failure when the S3 backup apparently should be disabled.

And, we seem to have an orphaned pod (I'll see if I can lay it to rest):
```
Error: failed to cleanup and remove container 1baab81a50cb4118da51767664715b7ce1b6378f999d5e7e97e0a74794b453af: container 1baab81a50cb4118da51767664715b7ce1b6378f999d5e7e97e0a74794b453af is the infra container of pod 706f45cdb739c4e73b19687a0e33779362096edc869a65d60cf94b1dad4b0a0d and cannot be removed without removing the pod
```

**Comment by dbutenhof** [comment]
> There's a ginormous traceback (or multiple tracebacks...it's really hard to tell) which seems to center around `psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "templates_name_key"`, which is I believe a timing-based failure that Dave is working on. So, I'll resubmit the job and see if it passes this time.

I fixed the problem, but I'm not sure whether this branch is merged up to date. In any case, ignoring the duplicate template doesn't appear to stop SQLAlchemy (or maybe the engine code in `psycopg2`) from doing a stack dump. I haven't figured out why; but, presuming the branch has the fix, it's innocuous. Just really annoying...

> Hey @dbutenhof, is this "S3 upload" failure expected? (It doesn't seem to be causing any problems, it just jumped out at me as I was looking at the log, so I thought I'd ask.)

We don't have S3 backup configured, so, yeah, this will always appear.

**Comment by portante** [comment]
> The changes look fine. Unfortunately, one of the Pbench Server functional tests failed, which (presumably) has nothing to do with this PR.
> 
> There's a ginormous traceback (or multiple tracebacks...it's really hard to tell) which seems to center around `psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "templates_name_key"`, which is I believe a timing-based failure that Dave is working on. So, I'll resubmit the job and see if it passes this time.

If this PR is rebased on `main` those stack traces should go away.

> ```
> Dec 09 13:08:01 pbenchinacan pbench-backup-t[729]: 2022-12-09T13:08:01.894711 WARNING 729 140579956992896 pbench-backup-tarballs pbench-backup-tarballs sanity_check 71 -- Bucket not defined in config file - S3 backup is disabled.
> ```
> 
> But, it's interesting that we note a failure when the S3 backup apparently should be disabled.

This is a very old message that we added for our current production instance while we were working out S3 backups to an internal S3 instance.  Once we move to an object store, this code should be removed, since the object store maintenance should handle its own backups.

**Comment by dbutenhof** [comment]
> E               AssertionError: Exceeded timeout: state Indexed, status {'backup': 'ok', 'index': 'ok'}

Ah. This is a failure I only started seeing recently. I just saw it in the audit commit I pushed yesterday, and I realize that I'd fixed it (I think) in the "gateway" fix which I now believe is a dead end in favor of Pete's preferred proxy solution.

If my new audit commit works, I'll push a separate PR with what I hope is a fix, in isolation.

**Comment by webbnh** [comment]
🎉  The tests passed this time.  Now we just need another review approval....

**Comment by dbutenhof** [comment]
> tada The tests passed this time. Now we just need another review approval....

Yeah; if I'm right (and I think I am), it's a race between BACKUP and UNPACK, which can find the second with an empty operations list... which fails to write the operation status "ok" that the PUT test is expecting.

**Review by dbutenhof — APPROVED** [review]
While I'm tempted to make some lame alt text joke like "This is not the elk you're looking for ...", yeah, this is OK.

**Pull request merged**